### DISCUSSION TITLE
Faustjs - doesn't work on Amplify AWS. What to do?

### DISCUSSION THREAD

**Discussion by gedlow** [wpengine/faustjs#1679] [Q&A]
## Environment
- `@faustwp/core` version: 1.2.0
- `@faustwp/cli` version: 1.2.1
- `@faustwp/blocks` version: NA
- `@faustwp/block-editor-utils` version: NA
- WordPress version: 6.4.2
- Browser, OS, and other relevant software versions: hosted on amplify / aws 

## Description
Faustjs doesn't seem to work on Amplify (AWS). From what I can tell, the problem is around Apollo Client and how it fetches the queries.

## Steps to Reproduce

1. Clone and set up an example project from https://github.com/wpengine/faustjs/tree/canary/examples/next/faustwp-getting-started
2. Upload on Amplify and configure envs;
3. Launch the amplify link.

## Expected vs. Actual Behavior
Expected: a website to work;
Actual: nextjs throws 500

## Reproducible Demo

here's the link to the demo:  https://master.dhl1hn8ryoygs.amplifyapp.com/

**Comment by theodesp** [comment]
Hey @gedlow I had a loot at this. It looks like there is  GET query during this error and this fails for some reason with

```
networkError: TypeError: Only absolute URLs are supported
```
The query url was:
```
chosenURI: /index.php?graphql&query=query%20GetSeedNode(%24id%3A%20ID!%20%3D%200%2C%20%24uri%3A%20String!%20%3D%20%22%22%2C%20%24asPreview%3A%20Boolean%20%3D%20false)%20%7B%0A%20%20...%20on%20RootQuery%20%40skip(if%3A%20%24asPreview)%20%7B%0A%20%20%20%20nodeByUri(uri%3A%20%24uri)%20%7B%0A%20%20%20%20%20%20__typename%0A%20%20%20%20%20%20...GetNode%0A%20%20%20%20%7D%0A%20%20%20%20__typename%0A%20%20%7D%0A%20%20...%20on%20RootQuery%20%40include(if%3A%20%24asPreview)%20%7B%0A%20%20%20%20contentNode(id%3A%20%24id%2C%20idType%3A%20DATABASE_ID%2C%20asPreview%3A%20true)%20%7B%0A%20%20%20%20%20%20__typename%0A%20%20%20%20%20%20...GetNode%0A%20%20%20%20%7D%0A%20%20%20%20__typename%0A%20%20%7D%0A%7D%0A%0Afragment%20GetNode%20on%20UniformResourceIdentifiable%20%7B%0A%20%20__typename%0A%20%20uri%0A%20%20id%0A%20%20...DatabaseIdentifier%0A%20%20...ContentType%0A%20%20...User%0A%20%20...TermNode%0A%20%20...ContentNode%0A%20%20...MediaItem%0A%20%20...Page%0A%7D%0A%0Afragment%20DatabaseIdentifier%20on%20DatabaseIdentifier%20%7B%0A%20%20databaseId%0A%20%20__typename%0A%7D%0A%0Afragment%20MediaItem%20on%20MediaItem%20%7B%0A%20%20id%0A%20%20mimeType%0A%20%20__typename%0A%7D%0A%0Afragment%20ContentType%20on%20ContentType%20%7B%0A%20%20name%0A%20%20isFrontPage%0A%20%20isPostsPage%0A%20%20__typename%0A%7D%0A%0Afragment%20Page%20on%20Page%20%7B%0A%20%20isFrontPage%0A%20%20isPostsPage%0A%20%20__typename%0A%7D%0A%0Afragment%20TermNode%20on%20TermNode%20%7B%0A%20%20isTermNode%0A%20%20slug%0A%20%20taxonomyName%0A%20%20__typename%0A%7D%0A%0Afragment%20ContentNode%20on%20ContentNode%20%7B%0A%20%20isContentNode%0A%20%20slug%0A%20%20contentType%20%7B%0A%20%20%20%20node%20%7B%0A%20%20%20%20%20%20name%0A%20%20%20%20%20%20__typename%0A%20%20%20%20%7D%0A%20%20%20%20__typename%0A%20%20%7D%0A%20%20template%20%7B%0A%20%20%20%20templateName%0A%20%20%20%20__typename%0A%20%20%7D%0A%20%20__typename%0A%7D%0A%0Afragment%20User%20on%20User%20%7B%0A%20%20name%0A%20%20userId%0A%20%20databaseId%0A%20%20__typename%0A%7D&operationName=GetSeedNode&variables=%7B%22id%22%3A0%2C%22uri%22%3A%22%2F%22%2C%22asPreview%22%3Afalse%7D
```

Looking deeper into AWS I found out that by default Amplify does not expose ENV variables in the SSR context

https://docs.aws.amazon.com/amplify/latest/userguide/ssr-environment-variables.html

So to fix that you need to manually add Faust ENV variables them to the **build configuration** the `.env.production`:

<img width="1748" alt="Screenshot 2023-12-14 at 14 15 22" src="https://github.com/wpengine/faustjs/assets/328805/c3b4499e-2e1a-4aa5-bfb4-21acba2e2742">

Once you do that then you should be able to see the page:

<img width="1786" alt="Screenshot 2023-12-14 at 14 15 00" src="https://github.com/wpengine/faustjs/assets/328805/189679cd-21a3-43a1-b4de-9f2115e8d57f">

I'm moving this issue to discussions for reference.

**Reply by gedlow** [reply]
that worked, thanks for your help @theodesp
